[CI] AssertionError in GoogleCloudStorageBlobContainerRetriesTests testWriteLargeBlob

Failure on 7.7: https://gradle-enterprise.elastic.co/s/6q5voyyn7k3vc

I couldn't reproduce locally (even leavign out the -Dbuild.snapshot=false):
```
./gradlew ':plugins:repository-gcs:test' --tests "org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobContainerRetriesTests.testWriteLargeBlob" \
  -Dtests.seed=74CA5748F999C5E9 \
  -Dtests.security.manager=true \
  -Dbuild.snapshot=false \
  -Dtests.jvm.argline="-Dbuild.snapshot=false" \
  -Dtests.locale=ca-ES \
  -Dtests.timezone=Asia/Tomsk \
  -Dcompiler.java=14
```

There is an AssertionError in the logs like this:
```
java.lang.AssertionError: 
Expected: (<15728640> or <7388234>)
     but: was <1949696>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18) ~[hamcrest-2.1.jar:2.1]
	at org.junit.Assert.assertThat(Assert.java:956) ~[junit-4.12.jar:4.12]
	at org.junit.Assert.assertThat(Assert.java:923) ~[junit-4.12.jar:4.12]
	at org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobContainerRetriesTests.lambda$testWriteLargeBlob$12(GoogleCloudStorageBlobContainerRetriesTests.java:411) ~[test/:?]
	at org.elasticsearch.repositories.blobstore.ESMockAPIBasedRepositoryIntegTestCase.lambda$wrap$4(ESMockAPIBasedRepositoryIntegTestCase.java:238) ~[framework-7.7.0.jar:7.7.0]
	at org.elasticsearch.repositories.gcs.GoogleCloudStorageBlobContainerRetriesTests.lambda$safeHandler$13(GoogleCloudStorageBlobContainerRetriesTests.java:452) ~[test/:?]
	at com.sun.net.httpserver.Filter$Chain.doFilter(Filter.java:79) [?:1.8.0_241]
```

I found https://github.com/elastic/elasticsearch/issues/52906 but the errors in the log look different so I opted for opening a new issue, please close or continue the investigation in that one if after all the issues seem to be related.

--- a/plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobContainerRetriesTests.java
+++ b/plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobContainerRetriesTests.java
@@ -51,6 +51,7 @@ import org.elasticsearch.rest.RestUtils;
 import org.elasticsearch.test.ESTestCase;
 import org.junit.After;
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.threeten.bp.Duration;
 
 import java.io.IOException;
@@ -98,6 +99,11 @@ public class GoogleCloudStorageBlobContainerRetriesTests extends ESTestCase {
         assertThat(httpServer, notNullValue());
         InetSocketAddress address = httpServer.getAddress();
         return "http://" + InetAddresses.toUriString(address.getAddress()) + ":" + address.getPort();
+    }
+
+    @BeforeClass
+    public static void skipJava8() {
+        GoogleCloudStorageBlobStoreRepositoryTests.assumeNotJava8();
     }
 
     @Before

--- a/plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStoreRepositoryTests.java
+++ b/plugins/repository-gcs/src/test/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStoreRepositoryTests.java
@@ -50,6 +50,7 @@ import org.elasticsearch.repositories.RepositoriesService;
 import org.elasticsearch.repositories.Repository;
 import org.elasticsearch.repositories.blobstore.BlobStoreRepository;
 import org.elasticsearch.repositories.blobstore.ESMockAPIBasedRepositoryIntegTestCase;
+import org.junit.BeforeClass;
 import org.threeten.bp.Duration;
 
 import java.io.IOException;
@@ -69,11 +70,16 @@ import static org.elasticsearch.repositories.gcs.GoogleCloudStorageRepository.CL
 @SuppressForbidden(reason = "this test uses a HttpServer to emulate a Google Cloud Storage endpoint")
 public class GoogleCloudStorageBlobStoreRepositoryTests extends ESMockAPIBasedRepositoryIntegTestCase {
 
-    private static void assumeNotJava8() {
+    public static void assumeNotJava8() {
         assumeFalse("This test is flaky on jdk8 - we suspect a JDK bug to trigger some assertion in the HttpServer implementation used " +
             "to emulate the server side logic of Google Cloud Storage. See https://bugs.openjdk.java.net/browse/JDK-8180754, " +
             "https://github.com/elastic/elasticsearch/pull/51933 and https://github.com/elastic/elasticsearch/issues/52906 " +
             "for more background on this issue.", JavaVersion.current().equals(JavaVersion.parse("8")));
+    }
+
+    @BeforeClass
+    public static void skipJava8() {
+        assumeNotJava8();
     }
 
     @Override
@@ -123,7 +129,6 @@ public class GoogleCloudStorageBlobStoreRepositoryTests extends ESMockAPIBasedRe
     }
 
     public void testDeleteSingleItem() {
-        assumeNotJava8();
         final String repoName = createRepository(randomName());
         final RepositoriesService repositoriesService = internalCluster().getMasterNodeInstance(RepositoriesService.class);
         final BlobStoreRepository repository = (BlobStoreRepository) repositoriesService.repository(repoName);
@@ -170,62 +175,7 @@ public class GoogleCloudStorageBlobStoreRepositoryTests extends ESMockAPIBasedRe
         assertEquals("failed to parse value [101mb] for setting [chunk_size], must be <= [100mb]", e.getMessage());
     }
 
-    @Override
-    public void testSnapshotAndRestore() throws Exception {
-        assumeNotJava8();
-        super.testSnapshotAndRestore();
-    }
-
-    @Override
-    public void testList() throws IOException {
-        assumeNotJava8();
-        super.testList();
-    }
-
-    @Override
-    public void testIndicesDeletedFromRepository() throws Exception {
-        assumeNotJava8();
-        super.testIndicesDeletedFromRepository();
-    }
-
-    @Override
-    public void testDeleteBlobs() throws IOException {
-        assumeNotJava8();
-        super.testDeleteBlobs();
-    }
-
-    @Override
-    public void testWriteRead() throws IOException {
-        assumeNotJava8();
-        super.testWriteRead();
-    }
-
-    @Override
-    public void testReadNonExistingPath() throws IOException {
-        assumeNotJava8();
-        super.testReadNonExistingPath();
-    }
-
-    @Override
-    public void testContainerCreationAndDeletion() throws IOException {
-        assumeNotJava8();
-        super.testContainerCreationAndDeletion();
-    }
-
-    @Override
-    public void testMultipleSnapshotAndRollback() throws Exception {
-        assumeNotJava8();
-        super.testMultipleSnapshotAndRollback();
-    }
-
-    @Override
-    public void testSnapshotWithLargeSegmentFiles() throws Exception {
-        assumeNotJava8();
-        super.testSnapshotWithLargeSegmentFiles();
-    }
-
     public void testWriteReadLarge() throws IOException {
-        assumeNotJava8();
         try (BlobStore store = newBlobStore()) {
             final BlobContainer container = store.blobContainer(new BlobPath());
             byte[] data = randomBytes(GoogleCloudStorageBlobStore.LARGE_BLOB_THRESHOLD_BYTE_SIZE + 1);

